contracts/bridge/contracts/common/fee-market/FeeMarket.sol

```diff
@@ -0,0 +1,448 @@
+pragma solidity >=0.6.0 <0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../../interfaces/IFeeMarket.sol";
+
+contract FeeMarket is IFeeMarket {
+    event SetOwner(address indexed o, address indexed n);
+    event SetOutbound(address indexed out, uint256 flag);
+    event SetParaTime(uint32 slash_time, uint32 relay_time);
+    event SetParaRelay(uint32 assigned_relayers_number, uint256 collateral);
+    event Slash(address indexed src, uint wad);
+    event Reward(address indexed dst, uint wad);
+    event Deposit(address indexed dst, uint wad);
+    event Withdrawal(address indexed src, uint wad);
+    event Locked(address indexed src, uint wad);
+    event UnLocked(address indexed src, uint wad);
+    event AddRelayer(address indexed prev, address indexed cur, uint fee);
+    event RemoveRelayer(address indexed prev, address indexed cur);
+    event OrderAssgigned(uint256 indexed key, uint256 timestamp, uint32 assigned_relayers_number, uint256 collateral);
+    event OrderSettled(uint256 indexed key, uint timestamp);
+
+    address private constant SENTINEL_HEAD = address(0x1);
+    address private constant SENTINEL_TAIL = address(0x2);
+    // System treasury
+    address public immutable VAULT;
+
+    // SlashAmount = CollateralPerorder * LateTime / SlashTime
+    uint32 public SlashTime;
+    // Time assigned relayer to relay messages
+    uint32 public RelayTime;
+    // Fee market assigned relayers numbers
+    uint32 public AssignedRelayersNumber;
+    // The collateral relayer need to lock for each order.
+    uint256 public CollateralPerorder;
```

Comment:
naming, start with lowercase.

Reply:
`camelCase` is good for storage variables